Additional resources which are crashing with Error: Plugin did not respond

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Problem description

Resources fail with Error: Plugin did not respond.
On Monday 2nd August 2021, eveything was working fine. We were working on some new stuff in our infrastructure and on Tuesday morning (3rd August) when we ran the deployments they started to fail with the Error: Plugin did not respond. We did not update the version of AzureRM nor Terraform. We were using TF 1.0.3 and Azurerm 2.64.0. 
We looked into the existing issues and we found that multiple resources are affected by this and we waiting for the 2.71.0 release hoping this would fix the raised issues.
We updated to Azurerm 2.71.0 and there are still some resources affected by this error (see the list below).


### Terraform (and AzureRM Provider) Version
* Terraform v1.0.4
* AzureRM v2.71.0

### Affected Resource(s)

* `azurerm_api_management `
*  `azurerm_monitor_diagnostic_setting`
* `azurerm_storage_account_sas`
* `azurerm_key_vault`
* `azurerm_kubernetes_cluster`
* `azurerm_servicebus_topic`

### Azure DevOps Agent output

```
Stack trace from the terraform-provider-azurerm_v2.71.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4a09ac5]

goroutine 225 [running]:
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers.resourceKubernetesClusterUpdate(0xc001acd780, 0x57e1000, 0xc0013ef880, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers/kubernetes_cluster_resource.go:1168 +0x685
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc000732d20, 0x6787328, 0xc001cb4240, 0xc001acd780, 0x57e1000, 0xc0013ef880, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:352 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000732d20, 0x6787328, 0xc001cb4240, 0xc00060f6c0, 0xc001b3f120, 0x57e1000, 0xc0013ef880, 0x0, 0x0, 0x0, ...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:464 +0x390
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0010b0438, 0x6787328, 0xc001cb4240, 0xc001790ff0, 0xc001cb4240, 0x5cc9180, 0xc001ff4100)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc001cb82e0, 0x67873d0, 0xc001cb4240, 0xc00060f490, 0xc001cb82e0, 0xc001ff4180, 0xc001203ba0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x5cc9180, 0xc001cb82e0, 0x67873d0, 0xc001ff4180, 0xc00009bb60, 0x0, 0x67873d0, 0xc001ff4180, 0xc001880000, 0xe1bd)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000170540, 0x67d3ab8, 0xc000511500, 0xc0013f4480, 0xc000c8da40, 0x9d7cfa0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000170540, 0x67d3ab8, 0xc000511500, 0xc0013f4480, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000c79410, 0xc000170540, 0x67d3ab8, 0xc000511500, 0xc0013f4480)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd

Error: The terraform-provider-azurerm_v2.71.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Example of Terraform Error log
```
│ Error: Plugin did not respond
│ 
│   with module.apimanagement.azurerm_api_management.ignis[0],
│   on Modules/ApiManagement/main.tf line 1, in resource "azurerm_api_management" "name":
│    1: resource "azurerm_api_management" "name" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.

╷
│ Error: Request cancelled
│ 
│   with module.databasebackupstorage.data.azurerm_storage_account_sas.db_backup_uploader_shared_access_signature,
│   on Modules/DBBackupStorageAccount/main.tf line 55, in data "azurerm_storage_account_sas" "db_backup_uploader_shared_access_signature":
│   55: data "azurerm_storage_account_sas" "db_backup_uploader_shared_access_signature" {
│ 
│ The plugin.(*GRPCProvider).ValidateDataResourceConfig request was cancelled.
```


### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform init`
2. `terraform apply`

--- a/internal/services/containers/kubernetes_cluster_resource.go
+++ b/internal/services/containers/kubernetes_cluster_resource.go
@@ -1165,7 +1165,7 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 			props.EnableRBAC = utils.Bool(rbacEnabled)
 
 			// Reset AAD profile is only possible if not managed
-			if props.AadProfile.Managed == nil || !*props.AadProfile.Managed {
+			if props.AadProfile == nil || props.AadProfile.Managed == nil || !*props.AadProfile.Managed {
 				log.Printf("[DEBUG] Updating the RBAC AAD profile")
 				future, err := clusterClient.ResetAADProfile(ctx, id.ResourceGroup, id.ManagedClusterName, *props.AadProfile)
 				if err != nil {
@@ -2017,7 +2017,7 @@ func flattenKubernetesClusterNetworkProfile(profile *containerservice.NetworkPro
 
 func expandKubernetesClusterRoleBasedAccessControl(input []interface{}, providerTenantId string) (bool, *containerservice.ManagedClusterAADProfile, error) {
 	if len(input) == 0 {
-		return false, &containerservice.ManagedClusterAADProfile{}, nil
+		return false, nil, nil
 	}
 
 	val := input[0].(map[string]interface{})

--- a/internal/services/containers/kubernetes_cluster_resource.go
+++ b/internal/services/containers/kubernetes_cluster_resource.go
@@ -2017,7 +2017,7 @@ func flattenKubernetesClusterNetworkProfile(profile *containerservice.NetworkPro
 
 func expandKubernetesClusterRoleBasedAccessControl(input []interface{}, providerTenantId string) (bool, *containerservice.ManagedClusterAADProfile, error) {
 	if len(input) == 0 {
-		return false, nil, nil
+		return false, &containerservice.ManagedClusterAADProfile{}, nil
 	}
 
 	val := input[0].(map[string]interface{})

